When an order is cancelled by a customer send an email to the shop.

cillian opened the pull request:
#### What? Why?

Fixes #6435 

> If the customer paid for their order by Stripe/Paypal then the Enterprise needs to know that the order was cancelled in order to arrange a refund.  Refunds are not automatically processed when an order is cancelled.

This will send a very basic email to the shop, it only includes a link to view the cancelled order in the admin area initially. Let me know if I did this too early, I understand it might need more planning first.

![cancel-email-for-shop](https://user-images.githubusercontent.com/101088/105510203-48d30100-5cc6-11eb-9295-225cdabc3921.png)

#### What should we test?

1. Find or set up an enterprise which allows customers to cancel orders (Admin > Shop Preferences > Change Orders)
2. Sign in as a customer and create an order (I'm not sure if guest checkout allows orders to be cancelled)
3. Cancel the order as the customer
4. Check enterprise owner receives email containing a link to the cancelled order in the admin area.

#### Release notes

When an order is cancelled by a customer send an email to the shop.

Changelog Category: User facing changes

#### Documentation updates

Add new email to https://github.com/openfoodfoundation/openfoodnetwork/wiki/Emails

Matt-Yorkley commented on config/locales/en.yml:
```diff
@@ -3630,6 +3630,11 @@ See the %{link} to find out more about %{sitename}'s features and to start using
+      cancel_email_for_shop:
+        greeting: "Dear %{name},"
+        subject: "Cancellation of Order"
+        intro: "An order has just been cancelled."
```
So this is only happening when an order is cancelled by a customer, right? I feel like we could add a bit more to the message here, like `A customer has cancelled their order` or something which conveys that context a bit.

mkllnk commented on app/services/customer_order_cancellation.rb:
```diff
@@ -6,9 +6,9 @@ def initialize(order)
-    if order.cancel
-      Spree::OrderMailer.cancel_email_for_shop(order).deliver_later
-    end
+    return if !order.cancel
```
Or
```suggestion
    return unless order.cancel
```

Matt-Yorkley replied:
:heavy_check_mark:

mkllnk reviewed (approved):
Nice. Can you rebase your PR and trigger a new build? It looks like there were some intermittent failures.

cillian commented:
@mkllnk @Matt-Yorkley Thanks, I have updated the text a bit now and rebased with the latest master as suggested. Although still getting some intermittent failures on SemaphoreCi that pass when run individually locally.

Matt-Yorkley reviewed (approved):
Great, thanks :+1:

filipefurtad0 commented:
Hi @cillian ,

Indeed, before this PR, the only way a hub manager could know of a customer-cancelled order and it's due credit, was through the backoffice. After this PR:

- As a customer: I placed an order through the shopfront; Checked order confirmation Email
- As the shop manager: I verified the order is in "Complete" and "Paid" state, under `/admin/orders`
- As a customer: cancelled the order - received the usual cancellation Email
- As a the shop manager: checked the order state is now "Cancelled" and "Credit Owed" state -> **a new email** arrives and notifies the shop manager, that the customer cancelled that order :+1:  

![image](https://user-images.githubusercontent.com/49817236/106281425-1d589500-6237-11eb-9b22-326752f74c7b.png)

I checked the link "View cancelled order", but the <href> leads the user to this address: 

`https://staging.coopcircuits.fr/admin/orders/<order_id>` -> error 404, page not found

when it rather should redirect the user to 
`https://staging.coopcircuits.fr/admin/orders/<order_id>/edit` -> just the **edit** bit missing here.


Other than this, this PR worked like a charm. This should be a really quick fix, right? Would you like to add an extra commit on this one @cillian ? Maybe not necessary to create a new issue?

filipefurtad0 commented:
Opened #6767 to address this last bit.

cillian commented:
Thanks @filipefurtad0 I fixed the link now, I just missed the issue the issue you created for that by a minute :-)

filipefurtad0 commented:
Great, thanks! :+1:

filipefurtad0 commented:
Alright,
Redirection now works - good to go!
